arrows/core/derive_metadata.cxx

```diff
@@ -0,0 +1,359 @@
+// Compute Video NIIRS image quality measure
+// Estimation based on General Image Quality Equation v5 (GIQE5)
+// See https://gwg.nga.mil/ntb/baseline/docs/GIQE-5_for_Public_Release.pdf
+double
+compute_vniirs( double gsd, double rer, double snr )
+{
+  // Taken from Table 2
+  constexpr double a0 = 9.57, a1 = -3.32, a2 = 3.32,
+                   a3 = -1.9, a4 = -2.0,  a5 = -1.8;
+
+  constexpr double meters_to_inches = 1.0 / 0.0254;
+  gsd *= meters_to_inches;
+
+  double const log10_gsd = std::log10( gsd );
+  double const log10_rer = std::log10( rer );
+
+  auto vniirs = a0 +
+                a1 * log10_gsd +
+                a2 * ( 1.0 - std::exp( a3 / snr ) ) * log10_rer +
+                a4 * std::pow( log10_rer, 4 ) +
+                a5 / snr;
+  return vniirs;
+}
+
+// ----------------------------------------------------------------------------
+double
+compute_rer()
+{
+  // TODO: Implement
+  return 0.3; // Dummy value within reasonable range
+}
+
+// ----------------------------------------------------------------------------
+double
+compute_snr()
+{
+  // TODO: Implement
+  return 15.0; // Dummy value within reasonable range
+}
+
+} // end namespace
+
+// ----------------------------------------------------------------------------
+kwiver::vital::metadata_vector
+compute_derived_metadata( kwiver::vital::metadata_vector metadata_vec,
+                          size_t frame_width, size_t frame_height )
+{
+  kv::metadata_vector updated_values;
+
+  for( auto const metadata : metadata_vec )
+  {
+    // Deep copy metadata
+    auto updated_metadata =
+      std::make_shared< kv::metadata >( metadata->deep_copy() );
+
+    try
+    {
+      // Compute slant range. Must be inserted before GSD calculation
+      auto const slant_range_value = compute_slant_range( updated_metadata );
+      auto const& slant_range_trait =
+        meta_traits.find( kv::VITAL_META_SLANT_RANGE );
+      updated_metadata->add(
+        slant_range_trait.create_metadata_item( slant_range_value ) );
+
+      // Compute GSD
+      auto const gsd_value =
+        compute_gsd( updated_metadata, frame_width, frame_height );
+      auto const& gsd_trait = meta_traits.find( kv::VITAL_META_AVERAGE_GSD );
+      updated_metadata->add( gsd_trait.create_metadata_item( gsd_value ) );
+
+      // Compute VNIIRS
+      auto const rer_value = compute_rer();
+      auto const snr_value = compute_snr();
+      auto const vniirs_value = compute_vniirs( gsd_value, rer_value,
+                                                snr_value );
+      auto const& vniirs_trait = meta_traits.find( kv::VITAL_META_VNIIRS );
+      updated_metadata->add(
+        vniirs_trait.create_metadata_item( vniirs_value ) );
+    }
+    catch ( kv::invalid_value const& e )
+    {
+      // Fail silently
+    }
+
+    updated_values.push_back( updated_metadata );
+  }
+
+  return updated_values;
+}
+
+} // namespace core
```

Comment:
If we are going to accept blank lines between namespace openings / closings because that's how uncrustify does it, shouldn't the style guide be changed to reflect this?

Reply:
Possibly 🙂. There was some notion of eventually "fixing" Uncrustify to be able to remove those, but meanwhile some people have been pushing to just accept them. So I guess the "answer" is that this is a good question that I'd be inclined to possibly raise to a larger audience. (But I suspect no one cares enough about the no-blanks style...)